dms/models/dms_file.py

```diff
@@ -527,48 +296,322 @@ def _check_name(self):
         for record in self:
             if not file.check_name(record.name):
                 raise ValidationError(_("The file name is invalid."))
-            files = record.sudo().directory_id.file_ids.name_get()
-            if list(
-                filter(
-                    lambda file: file[1] == record.name and file[0] != record.id, files
-                )
+            files = record.sudo().directory_id.file_ids
+            if files.filtered(
+                lambda file, record=record: file.name == record.name and file != record
             ):
-                raise ValidationError(_("A file with the same name already exists."))
+                raise ValidationError(
+                    _("A file with the same name already exists in this directory.")
+                )
 
     @api.constrains("extension")
     def _check_extension(self):
-        for record in self:
-            if (
-                record.extension
-                and record.extension in self._get_forbidden_extensions()
-            ):
-                raise ValidationError(_("The file has a forbidden file extension."))
+        if self.filtered(
+            lambda rec: rec.extension
+            and rec.extension in self._get_forbidden_extensions()
+        ):
+            raise ValidationError(_("The file has a forbidden file extension."))
 
     @api.constrains("size")
     def _check_size(self):
-        for record in self:
-            if record.size and record.size > self._get_binary_max_size() * 1024 * 1024:
-                raise ValidationError(
-                    _("The maximum upload size is %s MB.") % self._get_binary_max_size()
-                )
+        if self.filtered(
+            lambda rec: rec.size > self._get_binary_max_size() * 1024 * 1024
+        ):
+            raise ValidationError(
+                _("The maximum upload size is %s MB.") % self._get_binary_max_size()
+            )
 
     # ----------------------------------------------------------
-    # Create, Update, Delete
+    # CRUD/ORM Overrides
     # ----------------------------------------------------------
-
     def _inverse_content(self):
         updates = defaultdict(set)
         for record in self:
-            values = self._get_content_inital_vals()
+            values = self._get_content_initial_vals()
             binary = base64.b64decode(record.content or "")
             values = record._update_content_vals(values, binary)
             updates[tools.frozendict(values)].add(record.id)
-        with self.env.norecompute():
-            for vals, ids in updates.items():
-                self.browse(ids).write(dict(vals))
+        for vals, ids in updates.items():
+            self.browse(ids).write(dict(vals))
+
+    def copy(self, default=None):
```

Comment:
It's impossible to follow the real changes with the reorder you have made on the methods. Please undo it, and leave it to the refactor (or accept it as it is, as working in community sometimes mean this, to resign to certain things that are not totally of our taste on the mood of focusing on the real jobs, which is to migrate this).

I can't continue reviewing this file meanwhile.

Reply:
Got it, I'm currently working on reverting every changes that are not migration related (such as file renaming, comments and reordered fields/methods.